listenbrainz/webserver/login/provider.py

```diff
@@ -30,8 +30,11 @@ def get_user():
         'redirect_uri': url_for('login.musicbrainz_post', _external=True)
     }, decoder=lambda b: ujson.loads(b.decode("utf-8")))
     data = s.get('oauth2/userinfo').json()
-    user = db_user.get_or_create(data.get('sub'))
+    musicbrainz_id = data.get('sub')
+    user = db_user.get_or_create(musicbrainz_id)
```

Comment:
This'll need to look up users based on the `metabrainz_user_id ` too (otherwise there's no point in storing it :)). It'll have to fallback to using the `sub` where `musicbrainz_row_id` isn't set, but should still check that it isn't clobbering another user:

```SQL
SELECT *
  FROM "user"
 WHERE musicbrainz_row_id = :metabrainz_user_id
    OR (LOWER(musicbrainz_id) = LOWER(:sub) AND musicbrainz_row_id IS NULL);
```

Reply:
Not sure what you're supposed to do if there's already a user with that name having a different row ID. Since there's a UNIQUE constraint on the name. You can log the issue somewhere and change the new username to something unique (append an underscore or something). Hopefully it never happens in practice.